### ISSUE TITLE
file_target of accept share response has duplicate folder path

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#37883]
### Steps to reproduce
- Set system config `share_folder` to "Shares" so that received shares are delivered to the share receiver in a folder called "Shares".
- Set parameter "shareapi_auto_accept_share" of app "core" to "no" (i.e. make it so that users have to accept received shares)
- Share a file from "Alice" to "Brian"
- As "Brian", accept the share
- Look at the response to the accept request

A Gherkin test scenario like this does it:
```
  Background:
    Given the administrator has set the default folder for received shares to "Shares"
    And parameter "shareapi_auto_accept_share" of app "core" has been set to "no"
    And user "Alice" has been created with default attributes and without skeleton files
    And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"

  Scenario Outline: Creating a share of a file with a user, the default permissions are read(1)+update(2)+can-share(16)
    Given using OCS API version "<ocs_api_version>"
    And user "Brian" has been created with default attributes and without skeleton files
    When user "Alice" shares file "textfile0.txt" with user "Brian" using the sharing API
    Then the OCS status code should be "<ocs_status_code>"
    And the HTTP status code should be "200"
    And the fields of the last response to user "Alice" sharing with user "Brian" should include
      | share_with             | %username%            |
      | share_with_displayname | %displayname%         |
      | file_target            | /Shares/textfile0.txt |
      | path                   | /textfile0.txt        |
      | permissions            | share,read,update     |
      | uid_owner              | %username%            |
      | displayname_owner      | %displayname%         |
      | item_type              | file                  |
      | mimetype               | text/plain            |
      | storage_id             | ANY_VALUE             |
      | share_type             | user                  |
    When user "Brian" accepts the share "/textfile0.txt" offered by user "Alice" using the sharing API
    Then the OCS status code should be "<ocs_status_code>"
    And the HTTP status code should be "200"
    And the fields of the last response to user "Alice" sharing with user "Brian" should include
      | share_with             | %username%                   |
      | share_with_displayname | %displayname%                |
      | file_target            | /Shares/Shares/textfile0.txt |
      | path                   | /Shares/textfile0.txt        |
      | permissions            | share,read,update            |
      | uid_owner              | %username%                   |
      | displayname_owner      | %displayname%                |
      | item_type              | file                         |
      | mimetype               | text/plain                   |
      | storage_id             | ANY_VALUE                    |
      | share_type             | user                         |
    And the content of file "/Shares/textfile0.txt" for user "Brian" should be "ownCloud test text file 0"
    Examples:
      | ocs_api_version | ocs_status_code |
      | 1               | 100             |
      | 2               | 200             |
```

### Expected behaviour
`file_target` in the response should be `/Shares/textfile0.txt`

### Actual behaviour
`file_target` in the response is `/Shares/Shares/textfile0.txt`

(the `share_folder` value has been duplicated)

### Server configuration
Current oC10 core master.

**Comment by phil-davis** [comment]
Assigned myself to write a test scenario that demostrates the current behavior, and one that demonstrates the expected behavior.

**Comment by phil-davis** [comment]
Test scenarios have been added. I removed my assignment. I added this to the ownCloud 10 Server Development project - it's a bug that needs to be fixed.

**Comment by jvillafanez** [comment]
Quick check: DB stores the right value and the web UI shows the file in the right target. The bug seems to be just in the response, so it should be easy to fix.

**Comment by phil-davis** [comment]
Fixed by #37973
